Source/FullScreenPokemon.ts

```diff
@@ -3285,10 +3285,10 @@ module FullScreenPokemon {
                 dialog: MenuGraphr.IMenuDialogRaw | MenuGraphr.IMenuDialogRaw[] = other.dialog;
 
             thing.collidedTrigger = other;
-            this.animateCharacterPreventWalking(thing);
+            thing.FSP.animateCharacterPreventWalking(thing);
 
             if (!other.keepAlive) {
-                this.killNormal(other);
+                thing.FSP.killNormal(other);
```

Comment:
Wrong way. The solution should be to call this function with the correct scope, not to revert to `thing.FSP`.

Look into using `.call` or `.apply`.